From 1db4893d7dd4580dbcc5d890f43676b2f70d057a Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ti.com>
Date: Wed, 22 Jan 2014 11:15:49 +0200
Subject: [PATCH 181/285] OMAPDSS: SiI9022: remove silly code

Remove unneeded "return"s and an odd check at the end of driver remove
function.

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
Signed-off-by: Darren Etheridge <detheridge@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/video/omap2/displays-new/encoder-sil9022.c |   10 ----------
 1 files changed, 0 insertions(+), 10 deletions(-)

diff --git a/drivers/video/omap2/displays-new/encoder-sil9022.c b/drivers/video/omap2/displays-new/encoder-sil9022.c
index 6038c06..e0ebdec 100644
--- a/drivers/video/omap2/displays-new/encoder-sil9022.c
+++ b/drivers/video/omap2/displays-new/encoder-sil9022.c
@@ -452,8 +452,6 @@ static void sil9022_disconnect(struct omap_dss_device *dssdev,
 	}
 
 	in->ops.dpi->disconnect(in, &ddata->dssdev);
-	return;
-
 }
 
 static int sil9022_enable(struct omap_dss_device *dssdev)
@@ -505,7 +503,6 @@ static void sil9022_disable(struct omap_dss_device *dssdev)
 	in->ops.dpi->disable(in);
 
 	dssdev->state = OMAP_DSS_DISPLAY_DISABLED;
-	return;
 }
 
 static void sil9022_set_timings(struct omap_dss_device *dssdev,
@@ -523,7 +520,6 @@ static void sil9022_set_timings(struct omap_dss_device *dssdev,
 	dssdev->panel.timings = *sil9022_timings;
 
 	in->ops.dpi->set_timings(in, sil9022_timings);
-	return;
 }
 
 static void sil9022_get_timings(struct omap_dss_device *dssdev,
@@ -531,7 +527,6 @@ static void sil9022_get_timings(struct omap_dss_device *dssdev,
 {
 	struct panel_drv_data *ddata = to_panel_data(dssdev);
 	*timings = ddata->timings;
-	return;
 }
 
 static int sil9022_check_timings(struct omap_dss_device *dssdev,
@@ -827,11 +822,6 @@ static int sil9022_remove(struct i2c_client *client)
 
 	omap_dss_put_device(ddata->in);
 
-	if (!client->adapter) {
-		dev_err(&client->dev, "No HDMI Device\n");
-		return -ENODEV;
-	}
-
 	return 0;
 }
 
-- 
1.7.5.4

